pkgs/applications/audio/gtk-wave-cleaner/default.nix

```diff
@@ -0,0 +1,59 @@
+{
+  fetchFromGitHub,
+  stdenv, gnumake, autoconf, automake, pkgconfig,
+  libsndfile, gtk2, fftw,
+  pulseSupport ? stdenv.config.pulseaudio or false, alsaSupport ? true, 
+  libpulseaudio ? null, pulseaudio ? null, alsaLib ? null,
+  cliSupport ? true, perl ? null,
+  xdgSupport ? true, xdg_utils ? null
+}:
+
+assert pulseSupport || alsaSupport || (builtins.match ".*-darwin" stdenv.system != null);
+assert pulseSupport -> libpulseaudio != null && pulseaudio != null;
+assert alsaSupport  -> alsaLib != null;
+assert cliSupport   -> perl != null;
+assert xdgSupport   -> xdg_utils != null;
+
+stdenv.mkDerivation rec {
+
+  pname = "gtk-wave-cleaner";
+
+  version = "0.22-01";
+
+  src = fetchFromGitHub {
+    owner = "AlisterH";
+    repo = "gwc";
+    rev = "eade82cf01270421973f955d82b91db479c58dbd";
+    sha256 = "1argcaaank7ds16r4f9w193897pwg6y5yw5pd3jm2xkxhblkz6pd";
+  };
+
+  nativeBuildInputs = [
+    gnumake 
+    pkgconfig 
+    autoconf 
+    automake 
+  ];
+
+  preConfigure = "autoreconf -i";

```

Comment:
You can use `autoreconfHook` in `nativeBuildInputs` instead of this and `ac` and `am`.